auth/server/src/main/java/com/seagates3/policy/IAMPolicyValidator.java

```diff
@@ -20,25 +20,197 @@
+    try {
+
+      JSONObject obj = new JSONObject(jsonPolicy);
+      if (obj.toString().length() > MAX_IAM_POLICY_SIZE) {
+        LOGGER.error("Cannot exceed quota for PolicySize: " +
+                     MAX_IAM_POLICY_SIZE);
+        return responseGenerator.limitExceeded(
+            "Cannot exceed quota for PolicySize: " + MAX_IAM_POLICY_SIZE);
+      }
+      response = validatePolicyElements(obj);
+    }
+    catch (JSONException e) {
+      response = responseGenerator.malformedPolicy(
+          "This policy contains invalid Json - " + e.getMessage());
+      LOGGER.error("This policy contains invalid Json - ", e.getMessage());
+    }
+    catch (Exception e) {
+      response = responseGenerator.malformedPolicy(e.getMessage());
+      LOGGER.error("Exception in ValidatePolicy -  ", e);
+    }
+    return response;
+  }
+
+  /**
+   * This method validate the policy elements
+ * @param jsonObject
+ * @return null if elements are valid. Not null Response if elements
+ * are not valid
+ * @throws JSONException
+ */
+  ServerResponse validatePolicyElements(JSONObject jsonObject)
+      throws JSONException {
+    ServerResponse response = null;
+    if (!jsonObject.has(JsonDocumentFields.VERSION) ||
+        !jsonObject.has(JsonDocumentFields.STATEMENT)) {
+      response = responseGenerator.malformedPolicy("Syntax errors in policy.");
+      LOGGER.error("Missing required field Version or Statement");
+      return response;
+    }
+    LOGGER.debug("Checking for unknown fields in policy doc");
+    response = checkUnknownElements(jsonObject, policyElements);
+    if (response != null) return response;
+    Iterator<String> keys = jsonObject.keys();
+    while (keys.hasNext()) {
+      String key = keys.next();
+      if (JsonDocumentFields.VERSION.equals(key)) {
+        LOGGER.debug("Validating IAM policy version field");
+        response = validateVersion(
+            jsonObject.get(JsonDocumentFields.VERSION).toString());
+        if (response != null) return response;
+      } else if (JsonDocumentFields.STATEMENT.equals(key)) {
+        LOGGER.debug("Validating IAM policy statement field syntax");
+        response = validateStatementSyntax(jsonObject);
+        if (response != null) return response;
+
+        if (jsonObject.get(JsonDocumentFields.STATEMENT) instanceof JSONArray) {
+          LOGGER.debug("Validating each policy statement object.");
+          JSONArray arr =
+              (JSONArray)jsonObject.get(JsonDocumentFields.STATEMENT);
+          for (int count = 0; count < arr.length(); count++) {
+            JSONObject obj = (JSONObject)arr.get(count);
+            response = validateStatementElements(obj);
+            if (response != null) {
+              return response;
+            }
+          }
+        } else if (jsonObject.get(
+                       JsonDocumentFields.STATEMENT) instanceof JSONObject) {
+          LOGGER.debug("Validating policy statement object.");
```

Comment:
There are too many debug statements, add only one if required

Reply:
These are condition based debug statements, only one of them will get added in logs.